src/withPalette.js

```diff
@@ -0,0 +1,121 @@
+/* @flow */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { KEY } from './PaletteProvider';
+
+import type { PaletteInstance, ColorProfile, PaletteDefaults } from './types';
+
+type State = {
+  palette: ?PaletteInstance,
+  globalDefaults: ?PaletteDefaults,
+};
+
+/**
+ * Connect component to a material palette notification channel in order to obtain
+ * pallete instance when palette is resolved.
+ * Prop `palette` will be passed to wrapped component with either `null` or with palette instance.
+ * 
+ * If `mapPaletteToStyle` is specified, it will be evaluated when pallete is available and
+ * the results will be passed to a `style` prop to wrapped component.
+ */
+export default function withMaterialPalette(
+  mapPaletteToStyle: ?(palette: PaletteInstance) => {
```

Comment:
Right, I somehow wanted to catch wrong properties in that Object with flow for the user, but just skimming the `StyleSheet` library within RN, there are no proper Flow definitions for that. Let's keep it like that then